cmd/buildah/config.go

```diff
@@ -145,14 +146,22 @@ func updateConfig(builder *buildah.Builder, c *cli.Context) {
 		}
 	}
 	if c.IsSet("entrypoint") {
-		entrypointSpec := make([]string, 3)
-		entrypointSpec[0] = "/bin/sh"
-		entrypointSpec[1] = "-c"
-		entrypointSpec[2] = c.String("entrypoint")
-		if len(strings.TrimSpace(c.String("entrypoint"))) == 0 {
-			builder.SetEntrypoint(nil)
+		var entrypointJSON []string
+		err := json.Unmarshal([]byte(c.String("entrypoint")), &entrypointJSON)
+
+		if err != nil {
+			// it wasn't a valid json array, fall back to string
+			entrypointSpec := make([]string, 3)
+			entrypointSpec[0] = "/bin/sh"
+			entrypointSpec[1] = "-c"
+			entrypointSpec[2] = c.String("entrypoint")
+			if len(strings.TrimSpace(c.String("entrypoint"))) == 0 {
```

Comment:
Can we do this check before we do the json.Unmarshal.

Reply:
you are referring to the line 158, len check? the other array isn't used in valid JSON case.